ivy_tests/test_ivy/test_functional/test_core/test_searching.py

```diff
@@ -0,0 +1,212 @@
+"""Collection of tests for searching functions."""
+
+# global
+from hypothesis import given, strategies as st
+import numpy as np
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+import ivy.functional.backends.numpy as ivy_np
+
+# argmax
+@given(
+    array_shape=helpers.lists(
+        st.integers(1, 10), min_size="num_dims", max_size="num_dims", size_bounds=[1, 10]
+    ),
+    input_dtype=st.sampled_from(ivy_np.valid_numeric_dtypes),
+    data=st.data(),
+    as_variable=st.booleans(),
+    with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="argmax"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+)
+def test_argmax(
+    array_shape,
+    input_dtype,
+    data,
+    as_variable,
+    with_out,
+    num_positional_args,
+    native_array,
+    container,
+    instance_method,
+    fw,
+):
+    x = data.draw(
```

Comment:
we should not use `data.draw` in the test body, we should instead refactor the helper functions, and find a solution which uses `st.composite`. If the helper is not very general, then feel free to add a private helper function into this `test_searching.py` file. However, if the helper is general and potentially useful for other tests, then it should be added to the `helpers.py` file, so that others can also make use of it.